state-chain/pallets/cf-swapping/src/lib.rs

```diff
@@ -2625,14 +2614,11 @@ pub mod pallet {
 			}
 			.map(|estimation_output| {
 				// Get price in terms of `output_asset per input_asset`
-				let price = mul_div_floor(
-					estimation_output.into(),
-					U256::one() << PRICE_FRACTIONAL_BITS,
-					estimation_input,
-				);
+				let price =
+					Price::from_amounts_bounded(estimation_output.into(), estimation_input.into());
```

Comment:
Note that what used to be a sequence * and /, we have now replaced with more expensive sqrt only to immediately ^2 the result... This isn't performance critical code, but still feels wasteful. Maybe we should add a separate implementation of `from_amounts_abouded` for `Price`? (instead of having it call `SqrtPriceQ64F96::from_amounts_bounded` internally?

Not sure if `SqrtPriceQ64F96::from_amounts_bounded` is still needed then... maybe it is an optimisation, but IMO it doesn't make a big difference to us, could consider replacing it with `Price::from_amounts_bounded.into()`.